torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -746,9 +746,67 @@ def call_function(
             {},
             "torch.ops.higher_order.map",
             source_target=self.value,
+            manually_set_subgraph_inputs=False,
             should_flatten_outputs=True,
         )
 
+        # Find out used lifted args in body_graph.
+        # Note: Those un-used lifted freevars are first dim variables used for
+        # sepculate subgraphs. They're lifted to parent when tracing, which
+        # causes their parent to lift to their graph inputs (they're not used anywhere in the parent
+        # since it's a temporary slice from parent graph's point of view) and
+        # finally becomes a get_item call in top-level grah.
+        used_lifted_freevars = {
```

Comment:
Talked to @zou3519  offline. This gluing code is hard to maintain. Will try the flatten-unflatten approach + dce to avoid gluing code like this.